### PR TITLE
WIP: UIP-47: update JS deps

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

WIP for updating the narrative JS deps.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/UIP-47
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [ ] Tests pass locally and in GitHub Actions
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_ - N/A

# Dev Checklist:

- [ ] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [ ] Any dependent changes have been merged and published in downstream modules

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,77 +8,77 @@
...
         "karma-safari-launcher": "^1.0.0",
-        "lint-staged": "^12.3.6",
+        "lint-staged": "^15.2.6",
         "load-grunt-tasks": "^5.1.0",
-        "msw": "0.35.0",
-        "postcss": "8.4.33",
-        "postcss-cli": "10.0.0",
-        "postcss-discard-comments": "6.0.0",
-        "postcss-scss": "^4.0.6",
+        "msw": "^2.3.1",
```
Comment (briehl) [inline package.json:59]: We can't update that one yet until we rework how it gets used, as versions after 0.35.0 don't provide AMD files. Ideally, we use something different and remove it or rework the tests that make use of it entirely. It's not used in production things.

File: test/unit/karma.conf.js [hunk 2]
```diff
@@ -102,7 +101,7 @@ module.exports = function (config) {
         exclude: [...alwaysExclude, ...isolatedTests],
         // test results reporter to use
         // available reporters: https://npmjs.org/browse/keyword/karma-reporter
-        reporters: ['mocha', 'coverage', 'json-result'],
+        reporters: ['mocha', 'coverage'],
```
Comment (briehl) [inline test/unit/karma.conf.js:104]: 👍  I'm not sure that json-result gets used anywhere anymore. Maybe it was put in for some debugging?

**Review by briehl — CHANGES_REQUESTED** [review]
We can't update msw yet.

File: package.json (hunk 1 above)
Comment (ialarmedalien) [reply package.json]: Yeah, I found that out the hard way. 😖

File: test/unit/karma.conf.js (hunk 2 above)
Comment (ialarmedalien) [reply test/unit/karma.conf.js:104]: I know I put it in for debugging and I think I had something in the GHA involving it, but I don't think it gets used now. I certainly don't use it!

**Review by briehl — APPROVED** [review]

**Pull request merged**